code-challenges.test.js

```diff
@@ -64,4 +203,70 @@ const accountTransactions2 = [250, -89, 100, -96]
 const accountTransactions3 = []
 // Expected output: []
 
-// b) Create the function that makes the test pass.
+
+// describe('newSum', () => {
+//   it('takes an array, adds the numbers inside, returns the sum', () => {
+//     const accountTransactions1 = [100, -17, -23, -9]
+//     // Expected output: [100, 83, 60, 51]
+//     const accountTransactions2 = [250, -89, 100, -96]
+//     // Expected output: [250, 161, 261, 165]
+//     const accountTransactions3 = "[]"
+//     // Expected output: [] 
+//   expect(newsSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+//   expect(newSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+//   expect(newSum(accountTransactions3)).toEqual([[]])
+//   })
+// })
+
+
+//RED- Good Failure
+// FAIL  ./code-challenges.test.js
+// newSum
+//   ✕ takes an array, adds the numbers inside, returns the sum (1 ms)
+
+// ● newSum › takes an array, adds the numbers inside, returns the sum
+
+//   ReferenceError: newsSum is not defined
+
+//     198 |     const accountTransactions3 = "[]"
+//     199 |     // Expected output: [] 
+//   > 200 |   expect(newsSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+//         |   ^
+//     201 |   expect(newSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+//     202 |   expect(newSum(accountTransactions3)).toEqual([[]])
+//     203 |   })
+
+//     at Object.expect (code-challenges.test.js:200:3)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.161 s
+// Ran all test suites.
+// error Command failed with exit code 1.
+
+//b) Create the function that makes the test pass.
+
+//Pseudo Code:
+//input
+//output
+//process:
+//create a function to return an empty array if the index of the array is 0
+//take an array and iterate across it, allowing the numbers to accumulate.
+//create a new variable to store the sum
+//Add one index inside the string to the next index.
+//iterate across the array by using a for loop
+//create new variable that keeps the new value
+//take the iterated array and add number values together
+//return an array
```

Comment:
Your thought process is on the right track, but the execution of what you want to do is your biggest obstacle. Reviewing the Javascript lectures and material in the syllabus to break down and understand what each concept does for you will definitely help to overcome this obstacle!